File: frontend/lib/production/tech-card-display.ts
/** Live display `{stored}/{N}` (Stage 28). Stored number stays `{orderNo}-{seq}`. */
export function formatTechCardDisplayNumber(
  storedNumber: string,
  plannedCount: number | null | undefined,
): string {
  const number = storedNumber.trim();
  if (plannedCount != null && plannedCount >= 1) {
    return `${number}/${plannedCount}`;
  }
  return number;
}

export function techCardVisibleNumber(card: {
  number: string;
  display_number?: string | null;
  tech_cards_planned_count?: number | null;
}): string {
  const display = card.display_number?.trim();
  if (display) return display;
  return formatTechCardDisplayNumber(card.number, card.tech_cards_planned_count);
}

File: backend/app/services/tech_card_display.py
"""Live display number for technical cards (Stage 28 / SL-STANDALONE-TC-v1).

Stored `TechnicalCard.number` stays `{orderNo}-{seq}`. UI/print/list append `/{N}`
from the live planned count on SalesOrder or standalone order group.
"""

from __future__ import annotations


def format_tech_card_display_number(
    stored_number: str, planned_count: int | None
) -> str:
    number = stored_number.strip()
    if planned_count is not None and planned_count >= 1:
        return f"{number}/{planned_count}"
    return number
